ports/esp32/machine_ulp.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 "Andreas Valder" <andreas.valder@serioese.gmbh>
```

Comment:
Just to check: please verify that you have the authority to release this code under MIT with your copyright (eg you didn't do it during employed hours), and that you didn't copy any of the code from somewhere else.

Reply:
I have done it during employed hours and we are using the code internally. Upstreaming our changes to open source projects is part of our policy and I get paid to do ao. None of it is copied.

Reply:
Ok, that sounds fine, thanks.